### PR TITLE
Fix parsing of floating point defaults in python_arg_parser

### LINKED ISSUE #8484: float default values for Scalar in native_functions.yaml don't work
`rrelu` is affected by this and (probably) gives incorrect behavior (I have not verified this)

```
func: rrelu(Tensor self, Scalar lower=0.125, Scalar upper=0.3333333333333333, bool training=false, Generator* generator=nullptr) -> Tensor
variants: function

func: rrelu_(Tensor self, Scalar lower=0.125, Scalar upper=0.3333333333333333, bool training=false, Generator* generator=nullptr) -> Tensor
variants: function
```

I gdb'ed into `rrelu` and printed out "lower":
```
(gdb) p upper
$1 = {tag = at::Scalar::Tag::HAS_i, v = {d = 0, i = 0}, t = {pImpl =
    0x7fffe951f120 <at::UndefinedTensor::_singleton>}}
(gdb) p lower
$2 = {tag = at::Scalar::Tag::HAS_i, v = {d = 0, i = 0}, t = {
    pImpl = 0x7fffe951f120 <at::UndefinedTensor::_singleton>}}
```
they have the `at::Scalar::Tag::HAS_i` flag, so they're both being treated as integers. Sending them to `std::cout` gives 0.

### PR DESCRIPTION
**Pull request by li-roy**
Fixes #8484

### CODE REVIEW

File: torch/csrc/utils/python_arg_parser.cpp [hunk 1]
```diff
@@ -184,7 +184,8 @@ void FunctionParameter::set_default_str(const std::string& str) {
     } else {
       // we sometimes rely on integer-vs-float values, e.g. with arange.
       const auto as_integer = parse_as_integer(str);
-      default_scalar = at::Scalar(as_integer.value_or(atof(str.c_str())));
+      default_scalar = as_integer.has_value() ? at::Scalar(as_integer.value()) :
```
[the remark this answers is not in the document]
Comment (goldsborough) [reply torch/csrc/utils/python_arg_parser.cpp]: `value_or` of `optional<T>` returns `T`, so passing the float here truncated it to an int (my interpretation)

[the remark this answers is not in the document]
Comment (li-roy) [reply torch/csrc/utils/python_arg_parser.cpp]: Yeah, what Peter said

**Review by zou3519 — APPROVED** [review]
Good catch!
Add a test please -- you can check the behavior of "torch.rrelu" (I think that's the only function we have that uses floating point default values)

**Review by gchanan — APPROVED** [review]
nice!

**Pull request merged**